lib/src/async/stream_buffer.dart

```diff
@@ -91,15 +91,15 @@ class StreamBuffer<T> implements StreamConsumer<List<T>> {
       var subsize = leftToRead > listCap ? listCap : leftToRead;
       if (chunk is List) {
         ret.setRange(follower, follower + subsize,
-            (chunk as List<T>).getRange(_offset, _offset + subsize));
+            chunk.getRange(_offset, _offset + subsize));
       } else {
         ret[follower] = chunk;
       }
       follower += subsize;
       _offset += subsize;
       _counter -= subsize;
       leftToRead -= subsize;
-      if (chunk is! List || _offset >= (chunk as List).length) {
```

Comment:
Should we file a bug about `is!` not promoting to `List` here?

Reply:
@pq do you know if there's a bug on file for this already for the analyzer? I did a [quick search](https://github.com/dart-lang/sdk/issues?utf8=✓&q=is%3Aissue+is%3Aopen+%22is%21%22+infer+type) on "is! infer type"... guessing this must already exist though :)

Reply:
I'm not sure if this is being tracked anywhere.  It's possible this will get better w/ the type work for NNBD.  @stereotype441: do you happen to know?